### ISSUE TITLE
add UI font setting

### ISSUE THREAD

**Issue by psifertex** [Vector35/binaryninja-api#1510]
EDIT: Original report was root-caused to something a bit different, but one of the upshots is that the fact that the current settings only control fonts as used in graphs and logs but the UI font sizes are not controllable is a bit of a headache in some situations. Updated to be a request for the UI font size to be controlled by a different setting.

As reported in slack, @whitequark has an issue with font scaling resulting in broken behavior:

QT_SCALE_FACTOR less than 1 causes the UI to become badly broken.

EG: 

```QT_SCALE_FACTOR=0.5 /Applications/Binary\ Ninja-dev.app/Contents/MacOS/binaryninja```
![Screen Shot 2020-01-15 at 12 22 28 PM](https://user-images.githubusercontent.com/140215/72455967-c13ce300-3791-11ea-855e-b75ce6d70709.png)

**Comment by whitequark** [comment]
I think I figured out what's happening here; a complicated and unfortunated chain of events. First, let me state my environment:
  * KDE on Debian;
  * "Force fonts DPI" turned off, no DPI override in Xrdb;
  * `xdpyinfo` reports: `resolution:    276x277 dots per inch`
  * Environment variables set by KDE:
```
QT_SCREEN_SCALE_FACTORS=eDP1=2;DP1=2;DP2=2;VIRTUAL1=2;`
QT_AUTO_SCREEN_SCALE_FACTOR=0
```

What ticked me off is this strange combination of UI elements:
![Screenshot_20200115_175520](https://user-images.githubusercontent.com/54771/72458283-32938a80-37c0-11ea-9598-34b6af88eeb1.png)
The sizing is all wrong, implying that there's a HiDPI problem. And indeed there is. It looks like Xorg at some point changed their naming conventions, and instead of `eDP1` the laptop screen is now called `eDP-1`. If I change the environment variable appropriately, the UI looks much more appropriate:
![Screenshot_20200115_175631](https://user-images.githubusercontent.com/54771/72458367-5ce54800-37c0-11ea-882a-1059cb964ec9.png)
(Ok, those icons are ghastly, but that's minor).

It looks like Qt uses `QT_SCREEN_SCALE_FACTORS` to scale the UI raster/vector elements, and leaves the fonts alone when using that setting; this is probably why I never noticed the change. If `QT_SCREEN_SCALE_FACTORS` is set appropriately, then there are no issues with using the `QT_SCALE_FACTOR` workaround that I had for font scaling before. I wouldn't be surprised if using `QT_SCALE_FACTOR` to set the ultimate scale factor below one never worked in the first place, it just wasn't happening.

However, I think this issue shouldn't be closed just yet. Using `QT_SCALE_FACTOR` to work around the lack of UI font size configurability in binja isn't very good because it also scales all the raster/vector UI elements, and they look pretty bad. Using `QT_FONT_DPI` is slightly better because now vector UI elements aren't scaled, but raster ones still are, in particular the dock widget icons. Also, menu fonts end up tiny.
<hr>

**TL;DR:** the issue as filed is not a problem with binja, but it would be ideal to have the ability to change the UI font size directly anyway.

**Comment by ranweiler** [comment]
See also: #667

**Comment by psifertex** [comment]
Good catch, closing this in favor of that one.
